metron-platform/metron-parsing/metron-parsers-common/src/main/java/org/apache/metron/parsers/json/JSONMapParser.java

```diff
@@ -202,7 +204,7 @@ public void init() {
       }
       return Collections.unmodifiableList(parsedMessages);
     } catch (Throwable e) {
-      String message = "Unable to parse " + new String(rawMessage) + ": " + e.getMessage();
+      String message = "Unable to parse " + new String(rawMessage, StandardCharsets.UTF_8) + ": " + e.getMessage();
```

Comment:
This needs to be the underlying parser charset, I think. Otherwise we'll be trying to decode with the wrong thing and not outputting the message in a useful manner.

This seems like it occurs a couple times in these messages, so make sure I'm not misinterpreting it.

Reply:
I need to look into this a bit more. You're correct, that should be the readCharset also. I may be missing something, but our parser configure(...) method doesn't seem to get the high level parser configuration detail, e.g. all the options noted here that are siblings of the `parserConfig` key - https://github.com/apache/metron/tree/master/metron-platform/metron-parsing#parser-configuration. It seems better to make this option general - sibling to `parserConfig` rather than within `parserConfig`, but I don't think the API will allow it in its current form.

Reply:
Ok @justinleet - I posted an updated PR against your PR. Please review and see if that addresses this.

Reply:
Update - sorry, I made this comment on the wrong comment from you. I meant this regarding the parserrunnerimpl class.

Reply:
Both issues are now addressed. See https://github.com/justinleet/metron/pull/22